src/ble/CHIPBleServiceData.h

```diff
@@ -64,17 +68,34 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint8_t GetAdvertisementVersion() const
+    {
+        uint8_t advertisementVersion = static_cast<uint8_t>((DeviceDiscriminatorAndAdvVersion[1] >> AdvertisementVersionShiftBits) &
+                                                            kAdvertisementVersionMask);
+        return advertisementVersion;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint8_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion =
+            static_cast<uint8_t>((advertisementVersion & kAdvertisementVersionMask) << AdvertisementVersionShiftBits);
+        DeviceDiscriminatorAndAdvVersion[1] =
+            (DeviceDiscriminatorAndAdvVersion[1] & kAdvertisementVersionMask) | advertisementVersion;
```

Comment:
This is still making the assumption that kAdvertisementVersionMask can be used for both masking the advertisement and for masking the "not advertisement", which is really confusing.

Maybe this would all be a little clearer if we just made `kAdvertisementVersionMask` be `0xf0` and did the shifting before masking on set and the masking before shifting on get?

Reply:
updated